community/spatial-index/src/test/java/org/neo4j/gis/spatial/index/curves/SpaceFillingCurveTest.java

```diff
@@ -338,39 +356,209 @@ public void shouldGet2DHilbertSearchTilesForManyLevels()
+    {
+        double total = 0;
+        double min = Double.MAX_VALUE;
+        double max = 0;
+        int count = 0;
+
+        void add( double value )
+        {
+            total += value;
+            min = Math.min( min, value );
+            max = Math.max( max, value );
+            count++;
+        }
+
+        double avg()
+        {
+            return total / count;
+        }
+    }
+
+    @Test
+    public void shouldHaveReasonableCoveredArea()
+    {
+        final double minExtent = 0.000001;
+        final double maxAspect = 100.0;
+        final int xmin = -100;
+        final int xmax = 100;
+        final int ymin = -100;
+        final int ymax = 100;
+        final double rectangleStepsPerDimension = 9.789;
+        final double extensionFactor = 5;
+        String formatHeader1 = "Level  Depth Limitation Configuration                  Area Ratio              Ranges                  Depth";
+        String formatHeader2 = "                                                        avg    min    max       avg    min    max       avg    min    max";
+        String formatBody = "%5d  %-42s   %7.2f%7.2f%7.2f   %7.2f%7d%7d   %7.2f%7d%7d";
+
+        Envelope envelope = new Envelope( xmin, xmax, ymin, ymax );
+        // For all levels
+        for ( int level = 1; level <= HilbertSpaceFillingCurve2D.MAX_LEVEL; level++ )
+        {
+            log( "" );
+            log( formatHeader1 );
+            log( formatHeader2 );
+            for ( SpaceFillingCurveConfiguration config : new SpaceFillingCurveConfiguration[]{new StandardConfiguration( 1 ), new StandardConfiguration( 2 ),
+                    new StandardConfiguration( 3 ), new StandardConfiguration( 4 ), new PartialOverlapConfiguration( 1, 0.99, 0.1 ),
+                    new PartialOverlapConfiguration( 1, 0.99, 0.5 ), new PartialOverlapConfiguration( 2, 0.99, 0.1 ),
+                    new PartialOverlapConfiguration( 2, 0.99, 0.5 ), new PartialOverlapConfiguration( 3, 0.99, 0.1 ),
+                    new PartialOverlapConfiguration( 3, 0.99, 0.5 ), new PartialOverlapConfiguration( 4, 0.99, 0.1 ),
+                    new PartialOverlapConfiguration( 4, 0.99, 0.5 )} )
+            {
+                MonitorDoubleStats areaStats = new MonitorDoubleStats();
+                MonitorStats rangeStats = new MonitorStats();
+                MonitorStats maxDepthStats = new MonitorStats();
+                HilbertSpaceFillingCurve2D curve = new HilbertSpaceFillingCurve2D( envelope, level );
+
+                // For differently shaped rectangles
+                for ( double xExtent = minExtent; xExtent <= xmax; xExtent *= extensionFactor )
+                {
+                    for ( double yExtent = minExtent; yExtent <= ymax; yExtent *= extensionFactor )
+                    {
+                        // Filter out very thin rectangles
+                        double aspect = xExtent > yExtent ? xExtent / yExtent : yExtent / xExtent;
```

Comment:
This line I find quite hard to read, maybe some parentheses or a proper if-else-statement would make it more readable. Or you could change it to something like `max(xExtent,yExtent) / min(xExtent,yExtent)`